### ISSUE TITLE
Unable to communicate with QBittorrent.

### ISSUE THREAD

**Issue by ShadeBlack** [Radarr/Radarr#1761]
**Description:**
Radarr was showing no activity despite torrents being active and successfully delivering torrent to qbitttorent.  I *think* Radarr was struggling with the special characters in the torrent name and would fail when trying to build a file path.

**Version:**
Radarr: 0.2.0.778
qbittorrent: 3.3.1

**message**
> Unable to communicate with QBittorrent.: Cannot combine OsPaths of different platforms ('D:\Torrents\Completed\radarr\' + 'Spirited Away (2001) [1080p BluRay HEVC/x265]')

**exception**
> System.Exception: Cannot combine OsPaths of different platforms ('D:\Torrents\Completed\radarr\' + 'Spirited Away (2001) [1080p BluRay HEVC/x265]')
   at NzbDrone.Common.Disk.OsPath.op_Addition(OsPath left, OsPath right) in C:\projects\radarr-usby1\src\NzbDrone.Common\Disk\OsPath.cs:line 283
   at NzbDrone.Core.Download.Clients.QBittorrent.QBittorrent.GetItems() in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\QBittorrent\QBittorrent.cs:line 77
   at NzbDrone.Core.HealthCheck.Checks.DownloadClientCheck.Check() in C:\projects\radarr-usby1\src\NzbDrone.Core\HealthCheck\Checks\DownloadClientCheck.cs:line 32

I removed the torrent from the downloads list in qbit and then rerun the health check task.  Radarr was then able to popular the activity list.

**Comment by galli-leo** [comment]
Yeah, you need to setup remote path mapping. As far as I can tell, qbittorrent is on a windows machine and radarr is on a linux one? So you would have to tell radarr (in remote path mappings) that the Windows path 'D:\Torrents\Completed\' is equivalent to '/path/to/media/on/linux'.

**Comment by ShadeBlack** [comment]
Both were on the same windows machine.

**Comment by ShadeBlack** [comment]
You can reproduce this problem simply by renaming a torrent name in the qbittorrent queue to include a forward slash '/'

> Unable to retrieve queue and history items from QBittorrent: Cannot combine OsPaths of different platforms ('D:\Torrents\Completed\radarr\' + 'Batman The Dark Knight 2008 (1080p /x265 Joy).m4v')

**Comment by galli-leo** [comment]
Ah sorry, I misread the issue :(

**Comment by ltjdym** [comment]
same problem here but w nzbget. All on same nas. Sonaar can find nzbget fine

**Comment by RedshirtMB** [comment]
You said your QBittorrent version was 3.3.1. Did you mean 3.3.13? I had problems with Radarr and Sonarr after upgrading to  .13. Hunted around and reinstalled QBittorrent 3.3.12 and the problems went away.

**Comment by galli-leo** [comment]
@RedshirtMB Nah, this is an error with the Release Name containing a forward slash.

**Comment by ShadeBlack** [comment]
@RedshirtMB 3.3.1 'no updates available' - I assume .13 is a nightly/dev build?
@galli-leo I appreciate the re-open :)
--
From what I can tell its rare for a group to include illegal file path characters in the release name.  Sadly it's still a possibility.  Once I realised it's just a case of renaming it, it's less of a problem as that applies a quick fix, it's a more a problem for the automation if someone isn't actively keeping an eye on Radarr's activity.

**Comment by simplymathematics** [comment]
Windows only with illegal filename. Same problem.

**Comment by ztjuh** [comment]
Any solution?

**Comment by galli-leo** [comment]
@Taloth It seems like this issue still exists, although you said you would fix it here: https://github.com/Sonarr/Sonarr/pull/1032#issuecomment-167986439 Did you fix that?

If not, should we use the same OsPathKind for instantiation here: https://github.com/Sonarr/Sonarr/blob/9af57c6786eedd9beda4e1c6b8cdca20d165b622/src/NzbDrone.Common/Disk/OsPath.cs#L309 or just instantiate the OsPath ourselves inside the download client correctly (instead of adding a string)?

**Comment by Taloth** [comment]
@galli-leo Yes and no, the problem with 1032 was that those invalid paths were in fact magnet links for which the actual torrent wasn't downloaded yet, vawen fixed that by filtering those out. We didn't fix the colon specifically.
So the question is whether this problem is caused by magnet or torrent files whose release/directory name contains these characters. Such a folder name would be invalid for windows, so what kind of directory does transmission _actually_ create? Is it a multi-file torrent? etc

**Comment by galli-leo** [comment]
@Taloth A good point, I haven't been able to find a torrent that has '/' in the name (or '\'). I think it might just not use it in the folder name, since it won't allow me to change the output directory to something that does contain that.

@ztjuh @simplymathematics Or anyone else in this thread. Can you either send me the offending torrent file via E-Mail (leonardo.galli@bluewin.ch) or check what the actual download path is according to qBittorrent?

**Comment by azumukupoe** [comment]
Is it impossible for Radarr to automatically rename torrents?

**Comment by Bastien42** [comment]
my issue is that when Radarr grabs this movie from : 1337x
it triggers that error in the log you've already partially talked about that in another issue but here i give you a files that you can try to debug.

radarr 0.2.0.1358 running on windows 10 pro 1903 everything runs from the same computer all local.

so the isse is that it grabbed the torrent great but then the / in the naming of the folder structure of the torrents triggers that issue from what i understand thanks for your time and for this soft :)

**Comment by johnnyxmas** [comment]
The problem still ongoing with the latest version. It is definitely related to torrents that contain forward slashes in the folder or filenames.

**Comment by Bastien42** [comment]
yes i can confirm that

**Comment by SiemdeNijs** [comment]
-and till this very day, this issue still exist, with the only solution being removing the torrent/article from download clients.. and the entire activity que of the download client in question will not show up in activity/post processing when it receives this error.. Meaning that it's just a matter of luck if your torrent doesnt contain slashes, or else you'll entire system will do nothing at all, because it cannot for the love of God replace 1 character ?

This kills the automation process.

**Comment by 3thanfox** [comment]
THANK YOU!!  I've been tinkering with Radarr for days trying to figure out why I couldn't get it it work.  Turns out, it was the "/" in the torrent name.  Deleted the one torrent, poof - works.

I can confirm this issue still exists.  Windows 10, Radarr v 0.2.0.1450 with qBitTorrent v4.2.1 (64bit).  I've added "/" to Radarr's "must not contain" under Indexers -> Restrictions settings to NOT grab torrents with a "/" in the name.  So far it appears to be working.  This certainly isn't a FIX, as now it won't grab some potentially great torrents just because of this oversight, but at least Radarr doesn't unexpectedly quit working anymore (fingers crossed).
